### PR TITLE
Add HiddenTagFilter

### PR DESCRIPTION
**Pull request by MinchinWeb**
Use the `h:1` tag to hide items from being listed

Includes tests (existing tests were adjusted to test for the expected behavior)

### CODE REVIEW

File: topydo/lib/Filter.py [hunk 1]
```diff
@@ -154,6 +155,32 @@ def match(self, p_todo):
             return False
 
 
+class HiddenTagFilter(Filter):
+    def __init__(self, p_todos):
+        """
+        Constructor.
+
+        A filter which selects a number of Todo instances from a TodoList
```
Comment (bram85) [inline topydo/lib/Filter.py]: The documentation matches the preceding filter.

**Comment by MinchinWeb** [comment]
@bram85 : Fixes added!

**Comment by MinchinWeb** [comment]
Rebased to v0.10 release

File: topydo/lib/Filter.py [hunk 2]
```diff
@@ -154,6 +155,29 @@ def match(self, p_todo):
...
 
+class HiddenTagFilter(Filter):
+    def __init__(self, p_todos):
+        """
+        Constructor.
+
+        A filter which hides items marked with the approriate tag.
+        """
+        super().__init__()
+        self.todos = p_todos
```
Comment (bram85) [inline topydo/lib/Filter.py]: `self.todos` is not used, so I'd remove it from the constructor.

**Comment by bram85** [comment]
Looks good as far as I can see, bar a minor remark on an unused variable.

Personally I use the `h` tag to hide an item just for one day, for instance I write `h:2016-08-24` in a todo item and have a column defined in column mode with filter `+Project -h:today`. Could be a future extension.

**Comment by MinchinWeb** [comment]
@bram85 : I've removed the unused variable. I think we're good to go then!

**Comment by bram85** [comment]
Thanks :)

**Pull request merged**